acceptance/openstack/rds/v3/rds_test.go

```diff
@@ -56,89 +45,335 @@ func TestRdsLifecycle(t *testing.T) {
-	rdsReadReplica, err := createResult.Extract()
+	t.Log("UpdateDataIp")
+
+	ip, err := security.UpdateDataIp(client, security.UpdateDataIpOpts{
+		InstanceId: rds.Id,
+		NewIp:      "192.168.30.254",
+	})
 	th.AssertNoErr(t, err)
-	jobResponse, err := createResult.ExtractJobResponse()
+	err = instances.WaitForJobCompleted(client, 600, *ip)
 	th.AssertNoErr(t, err)
-	err = instances.WaitForJobCompleted(client, 1200, jobResponse.JobID)
+
+	netClient, err := clients.NewNetworkV1Client()
 	th.AssertNoErr(t, err)
-	t.Logf("Created RDSv3 Read Replica: %s", rdsReadReplica.Instance.Id)
 
-	defer func() {
-		t.Logf("Attempting to delete RDSv3 Read Replica: %s", rdsReadReplica.Instance.Id)
-		_, err := instances.Delete(client, rdsReadReplica.Instance.Id).ExtractJobResponse()
+	elasticIP := networking.CreateEip(t, netClient, 100)
+	t.Cleanup(func() {
+		networking.DeleteEip(t, netClient, elasticIP.ID)
+	})
+
+	t.Log("AttachEip")
+
+	err = instances.AttachEip(client, instances.AttachEipOpts{
+		InstanceId: rds.Id,
+		PublicIp:   elasticIP.PublicAddress,
+		PublicIpId: elasticIP.ID,
+		IsBind:     pointerto.Bool(true),
+	})
+	th.AssertNoErr(t, err)
+	t.Cleanup(func() {
+		err = instances.AttachEip(client, instances.AttachEipOpts{
+			InstanceId: rds.Id,
+			IsBind:     pointerto.Bool(false),
+		})
 		th.AssertNoErr(t, err)
-		t.Logf("RDSv3 Read Replica instance deleted: %s", rdsReadReplica.Instance.Id)
-	}()
+	})
+
+	t.Log("ChangeOpsWindow")
+
+	err = instances.ChangeOpsWindow(client, instances.ChangeOpsWindowOpts{
+		InstanceId: rds.Id,
+		StartTime:  "22:00",
+		EndTime:    "02:00",
+	})
+	th.AssertNoErr(t, err)
+
+	t.Log("SingleToHa")
+
+	time.Sleep(5 * time.Second)
+
+	ha, err := instances.SingleToHa(client, instances.SingleToHaOpts{
+		InstanceId:    rds.Id,
+		AzCodeNewNode: az,
+	})
+	th.AssertNoErr(t, err)
+	err = instances.WaitForJobCompleted(client, 600, *ha)
+	th.AssertNoErr(t, err)
+
+	t.Log("ChangeFailoverMode")
+
+	mode, err := instances.ChangeFailoverMode(client, instances.ChangeFailoverModeOpts{
+		InstanceId: rds.Id,
+		Mode:       "sync",
+	})
+	th.AssertNoErr(t, err)
+	err = instances.WaitForJobCompleted(client, 600, mode.WorkflowId)
+	th.AssertNoErr(t, err)
+
+	t.Log("ChangeFailoverStrategy")
+
+	err = instances.ChangeFailoverStrategy(client, instances.ChangeFailoverStrategyOpts{
+		InstanceId:     rds.Id,
+		RepairStrategy: "availability",
+	})
+	th.AssertNoErr(t, err)
+
+	// haRds, err := instances.List(client, instances.ListOpts{
+	// 	Id: rds.Id,
+	// })
+	// th.AssertNoErr(t, err)
+	//
+	// az2 := clients.EnvOS.GetEnv("AVAILABILITY_ZONE_2")
+	// if az2 == "" {
+	// 	az2 = "eu-de-03"
+	// }
+	// TODO: Action Forbidden
+	// follower, err := instances.MigrateFollower(client, instances.MigrateFollowerOpts{
+	// 	InstanceId: rds.Id,
+	// 	NodeId:     haRds.Instances[0].Nodes[0].Id,
+	// 	AzCode:     az2,
+	// })
+	// th.AssertNoErr(t, err)
+	// err = instances.WaitForJobCompleted(client, 600, *follower)
+	// th.AssertNoErr(t, err)
```

Comment:
if this api not working please add some description about what returns

Reply:
done